video.js

```diff
@@ -0,0 +1,196 @@
+   video.on('click', function() { playpause(); } );
+   $('.btnPlay').on('click', function() { playpause(); } );
+   var playpause = function() {
+      if(video[0].paused || video[0].ended) {
+         $('.btnPlay').addClass('paused');
+         video[0].play();
+      }
+      else {
+         $('.btnPlay').removeClass('paused');
+         video[0].pause();
+      }
+   };
+
+   //stop button clicked
+   $('.btnStop').on('click', function() {
+      $('.btnPlay').removeClass('paused');
+      updatebar($('.progress').offset().left);
+      video[0].pause();
+   });
+
+   //fullscreen button clicked
+   $('.btnFS').on('click', function() {
+      if($.isFunction(video[0].webkitEnterFullscreen)) {
+         video[0].webkitEnterFullscreen();
+      }
+      else if ($.isFunction(video[0].mozRequestFullScreen)) {
+         video[0].mozRequestFullScreen();
+      }
+      else {
+         alert('Your browsers doesn\'t support fullscreen');
+      }
+   });
+
+
+   //sound button clicked
+   $('.sound').click(function() {
+      video[0].muted = !video[0].muted;
+      $(this).toggleClass('muted');
+      if(video[0].muted) {
+         $('.volumeBar').css('width',0);
+      }
+      else{
+         $('.volumeBar').css('width', video[0].volume*100+'%');
+      }
+   });
+
+   //VIDEO EVENTS
+
+   //VIDEO PROGRESS BAR
+   //when video timebar clicked
+   var timeDrag = false;   /* check for drag event */
+   $('.progress').on('mousedown', function(e) {
+      timeDrag = true;
+      updatebar(e.pageX);
+   });
+   $(document).on('mouseup', function(e) {
+      if(timeDrag) {
+         timeDrag = false;
+         updatebar(e.pageX);
+      }
+   });
+   $(document).on('mousemove', function(e) {
+      if(timeDrag) {
+         updatebar(e.pageX);
+      }
+   });
+   var updatebar = function(x) {
+      var progress = $('.progress');
+
+      //calculate drag position
+      //and update video currenttime
+      //as well as progress bar
+      var maxduration = video[0].duration;
+      var position = x - progress.offset().left;
+      var percentage = 100 * position / progress.width();
+      if(percentage > 100) {
+         percentage = 100;
+      }
+      if(percentage < 0) {
+         percentage = 0;
+      }
+      $('.timeBar').css('width',percentage+'%');
+      video[0].currentTime = maxduration * percentage / 100;
+   };
+   //VOLUME BAR
+   var volumeDrag = false;
+   $('.volume').on('mousedown', function(e) {
+      volumeDrag = true;
+      video[0].muted = false;
+      $('.sound').removeClass('muted');
+      updateVolume(e.pageX);
+   });
+   $(document).on('mouseup', function(e) {
+      if(volumeDrag) {
+         volumeDrag = false;
+         updateVolume(e.pageX);
+      }
+   });
+   $(document).on('mousemove', function(e) {
+      if(volumeDrag) {
+         updateVolume(e.pageX);
+      }
+   });
+   var updateVolume = function(x, vol) {
+      var volume = $('.volume');
+      var percentage;
```

Comment:
it's generally considered best practice to declare any variables or functions above where they are called. for example, updateVolume gets invoked on line 16, but defined on line 153. This way, 6 months from now when you're looking through this, you don't have to scroll around looking for various declarations, they're all sitting at the top to begin with.

Good work on being so thorough, though!